pkg/kubelet/cm/dra/plugin/plugin.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	utilversion "k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// DRAPluginName is the name of the in-tree DRA Plugin.
+	DRAPluginName = "kubernetes.io/dra"
+)
+
+// draPlugins map keeps track of all registered DRA plugins on the node
+// and their corresponding sockets.
+var draPlugins = &PluginsStore{}
+
+// RegistrationHandler is the handler which is fed to the pluginwatcher API.
+type RegistrationHandler struct{}
+
+// NewPluginHandler returns new registration handler.
+func NewRegistrationHandler() *RegistrationHandler {
+	return &RegistrationHandler{}
+}
+
+// RegisterPlugin is called when a plugin can be registered.
+func (h *RegistrationHandler) RegisterPlugin(pluginName string, endpoint string, versions []string) error {
+	klog.InfoS("Register new DRA plugin", "name", pluginName, "endpoint", endpoint)
+
+	highestSupportedVersion, err := h.validateVersions("RegisterPlugin", pluginName, versions)
+	if err != nil {
+		return err
+	}
+
+	// Storing endpoint of newly registered DRA Plugin into the map, where plugin name will be the key
+	// all other DRA components will be able to get the actual socket of DRA plugins by its name.
+	draPlugins.Set(pluginName, &Plugin{
+		endpoint:                endpoint,
+		highestSupportedVersion: highestSupportedVersion,
+	})
+
+	return nil
+}
+
+// Return the highest supported version.
+func highestSupportedVersion(versions []string) (*utilversion.Version, error) {
```

Comment:
This is almost exactly same as `pkg/volume/csi/csi_plugin.go:highestSupportedVersion` (https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_plugin.go#L859) ideally we can extract the common logic out and put it in the something like `pkg/kubelet/pluginmanager/util`

Reply:
Agreed. This can be moved to a helper.

However, I don't see this as a blocker for the `alpha` release. We can address this as we move to beta and are more certain that `dra` is on a path to becoming a GA feature someday.

Reply:
+1, not a blocker, thanks for creating isuse to track it.